### PR TITLE
Hummingbird analytics

### PR DESCRIPTION
**Pull request by snark**
Hummingbird analytics adapter prototype

    This changeset provides an MVP of a Hummingbird analytics adapter, which
    gathers bid/no bid information for each auction on a page and sends it
    to an endpoint defined in a config object. Info gathered includes a few
    details on the environment (country, pod, screen size, etc.) and, for
    submitted bids, CPM and time-to-respond data enabling us to do some
    counterfactual exploration of our bidder setup.

### CODE REVIEW

**Comment by bradchoate** [comment]
Looks good to me beyond that one suggestion. I'd rename it to `mavenAnalyticsAdapter` since all the other adapters are named after their respective organization.

File: modules/hummingbirdAnalyticsAdapter.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                    auctionObj = {
+                        auctionId: id,
+                        contentItem: options.contentItem,
+                        correlator: window.hummingbirdCorrelator,
+                        countryCode: options.countryCode,
+                        mavenChannel: options.mavenChannel,
+                        path: location.pathname,
+                        pod: Number(options.pod),
+                        productionDomain: options.productionDomain,
+                        screenSize: options.screenSize,
```
[the remark this answers is not in the document]
Comment (snark) [reply modules/hummingbirdAnalyticsAdapter.js]: 👍

File: modules/hummingbirdAnalyticsAdapter.js [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+                    // code, path, pod, screen size, domain/channel, and
+                    // timeout period), leaving only the auction ID as
+                    // duplicated in our batched values. (NB: We want to push
+                    // this information over the wire, but should be able to
+                    // cut down on bandwidth by sending all this identical
+                    // information only once per batch.)
+                    auctionObj = {
+                        auctionId: id,
+                        contentItem: options.contentItem,
+                        correlator: window.hummingbirdCorrelator,
```
[the remark this answers is not in the document]
Comment (snark) [reply modules/hummingbirdAnalyticsAdapter.js]: "Correlator" is by analogy to GA -- note that it's not the _same_ correlator as GA's -- so I'll leave it named that, but I will switch to contentItemId.

File: modules/hummingbirdAnalyticsAdapter.js [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+                        productionDomain: options.productionDomain,
+                        screenSize: options.screenSize,
+                        status: 'inProgress',
+                        timeoutPeriod: undefined,
+                    };
+                    // Sanity check on POD
+                    if (isNaN(auctionObj.pod) || auctionObj.pod > 999) {
+                        auctionObj.pod = 0;
+                    };
+                    currentAuctions[id] = auctionObj;
```
[the remark this answers is not in the document]
Comment (snark) [reply modules/hummingbirdAnalyticsAdapter.js]: Sounds good.

File: modules/hummingbirdAnalyticsAdapter.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (snark) [reply modules/hummingbirdAnalyticsAdapter.js]: Actually, after discussion with Ed, we're going to keep this as A/B/C for now -- determining device in Owl is simple, and Ed thinks mingling tablet and phone may skew some of the data. We'll revisit in a future sprint.

**Comment by snark** [comment]
Merged manually; now in `stable`!

**Pull request closed without merging**